acmeair-webapp/src/main/resources/static/js/acmeair-common.js

```diff
@@ -67,6 +76,7 @@ function login() {
 				return;
 			}
 			dojo.cookie("loggedinuser", userString, {expires: 5});
+        syncCoupon(userString);
```

Comment:
initiating data sync from front end may crash the backend service, if many users log in at the same time. preferably a background scheduled task does this in a dedicated thread.

Reply:
Can we just add a cache for it?